controllers/druid/handler.go

```diff
@@ -831,6 +844,13 @@ func makeStatefulSet(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid, ls map
 	}, nil
 }
 
+// currently only checks for replica count mismatch, can be extended further
+func statefulSetIsEquals(obj1, obj2 object) bool {
+	o1 := obj1.(*appsv1.StatefulSet)
+	o2 := obj2.(*appsv1.StatefulSet)
+	return *o1.Spec.Replicas == *o2.Spec.Replicas
+}
+
```

Comment:
@himanshug in case we add spec here, wonder if this will  conflict on scaling sts https://github.com/druid-io/druid-operator/pull/97 

Regardless are you planning to keep only replicas for now or extend for spec too

Reply:
for now, I am leaving it at just replica count mismatch ... but we can add more stuff here based on real world needs.